### PR TITLE
Enable plugin Android instrumentation tests on Firebase Test Lab

### PR DESCRIPTION
**Pull request by collinjackson**
Android only for now. We could have parameters to let developers choose between Android and iOS but it's lower priority since we can use iOS simulators on Cirrus.

~Do we need to have this command be tested in order to land this? I'd like to get the Android Firebase Test Lab tests running on Cirrus and then continue working on mocks for Flutter plugin tools commands.~ Filed https://github.com/flutter/flutter/issues/38986 for additional test coverage as suggested by Amir.

This PR is required for landing https://github.com/flutter/plugins/pull/1866

### CODE REVIEW

File: lib/src/firebase_test_lab_command.dart [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+          workingDir: androidDirectory);
+
+      if (exitCode != 0) {
+        failingPackages.add(packageName);
+        continue;
+      }
+
+      final String scriptDirectory = p.join(packagesDir.path, '..', 'script');
+      exitCode = await runAndStream(
+          p.join(scriptDirectory, 'firebase_test_lab.sh'), <String>[],
```
Comment (amirh) [inline lib/src/firebase_test_lab_command.dart]: Is there a pending PR with this shell script? anything there that we can't just do in Dart here?

Comment (collinjackson) [reply lib/src/firebase_test_lab_command.dart]: Yep. Good call. I'll move it.

File: lib/src/firebase_test_lab_command.dart [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+          new Directory(p.join(example.path, 'android'));
+      if (!new File(p.join(androidDirectory.path, _gradleWrapper))
+          .existsSync()) {
+        print('ERROR: Run "flutter build apk" on example app of $packageName'
+            'before executing tests.');
+        missingFlutterBuild.add(packageName);
+        continue;
+      }
+
+      int exitCode = await runAndStream(
```
Comment (amirh) [inline lib/src/firebase_test_lab_command.dart]: Do we need to activate the service account and set the project one per each example? If not I'd consider extracting these to a `_configureFirebaseProject` (for readability) and calling it only once.

Comment (collinjackson) [reply lib/src/firebase_test_lab_command.dart]: Good suggestion, done.

**Comment by collinjackson** [comment]
> LGTM modulo nits.
> 
> As discussed - we should start requiring tests for new plugin_tools code, for your consideration whether to start enforcing it after this PR 😄

Agree. I filed https://github.com/flutter/flutter/issues/38986

File: lib/src/firebase_test_lab_command.dart [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        defaultsTo: 'plugins_android_test/$gitRevision/$buildId');
+  }
+
+  @override
+  final String name = 'firebase-test-lab';
+
+  @override
+  final String description = 'Runs the instrumentation tests of the example '
+      'apps on Firebase Test Lab.\n\n'
+      'Runs tests in test_instrumentation folder using the '
```
Comment (digiter) [inline lib/src/firebase_test_lab_command.dart]: "Instrumentation" is a quite widely used term, for example, WidgetTester is actually an instrumentation to the Flutter app, as we need to *pump* the widgets.

Therefore I think "instrumentation_adapter" is a better name to reflect the fact that we are propagating the Dart test result as native instrumentation test result. Hence here would change to "Runs tests in instrumentation_adapter folder using the instrumentation_adapter plugin".

File: lib/src/firebase_test_lab_command.dart [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+      for (File test in testsDir.listSync()) {
+        exitCode = await runAndStream(
+            p.join(androidDirectory.path, _gradleWrapper),
+            <String>[
+              'assembleDebug',
+              '-Pverbose=true',
+              '-Ptrack-widget-creation=false',
+              '-Pfilesystem-scheme=org-dartlang-root',
+              '-Ptarget=${test.path}'
```
Comment (digiter) [inline lib/src/firebase_test_lab_command.dart]: Are "-Ptrack-widget-creation=false" and "-Pfilesystem-scheme=org-dartlang-root" really useful?

File: lib/src/firebase_test_lab_command.dart (hunk 3 above)
Comment (collinjackson) [reply lib/src/firebase_test_lab_command.dart]: @amirh do you have a preference?

File: lib/src/firebase_test_lab_command.dart (hunk 4 above)
Comment (collinjackson) [reply lib/src/firebase_test_lab_command.dart]: I don't think so. I've removed them.

File: lib/src/firebase_test_lab_command.dart (hunk 3 above)
Comment (collinjackson) [reply lib/src/firebase_test_lab_command.dart]: Spoke with Amir offline, I'll go with instrumentation_adapter for now but I suspect we'll want to revisit this pattern before it's included in the Flutter engine.

**Comment by collinjackson** [comment]
Merging this to unblock flutter/plugins#1866

**Pull request merged**